bin/custom/questionsaddon/src/com/hybris/training/jobs/QuestionsUpdatesEmailSendingJob.java

```diff
@@ -0,0 +1,82 @@
+package com.hybris.training.jobs;
+
+import com.hybris.training.model.QuestionsUpdatesEmailProcessModel;
+import com.hybris.training.model.QuestionsUpdatesEmailSendingCronJobModel;
+import com.hybris.training.services.QuestionService;
+import de.hybris.platform.catalog.constants.CatalogConstants;
+import de.hybris.platform.core.model.user.CustomerModel;
+import de.hybris.platform.cronjob.enums.CronJobResult;
+import de.hybris.platform.cronjob.enums.CronJobStatus;
+import de.hybris.platform.jalo.JaloSession;
+import de.hybris.platform.jalo.SessionContext;
+import de.hybris.platform.processengine.BusinessProcessService;
+import de.hybris.platform.servicelayer.cronjob.AbstractJobPerformable;
+import de.hybris.platform.servicelayer.cronjob.PerformResult;
+import de.hybris.platform.servicelayer.user.UserService;
+import org.springframework.beans.factory.annotation.Required;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.Optional;
+
+public class QuestionsUpdatesEmailSendingJob extends AbstractJobPerformable<QuestionsUpdatesEmailSendingCronJobModel> {
+
+    private final static String PROCESS_DEFINITION_NAME = "questionsUpdatesEmailProcess";
+
+    private QuestionService questionService;
+    private BusinessProcessService businessProcessService;
+    private UserService userService;
+
+    @Override
+    public PerformResult perform(QuestionsUpdatesEmailSendingCronJobModel cronJobModel) {
+        QuestionsUpdatesEmailProcessModel emailProcessModel = businessProcessService
+                .createProcess(PROCESS_DEFINITION_NAME + System.currentTimeMillis(), PROCESS_DEFINITION_NAME);
+
+        configureSession(cronJobModel);
+
+        initProcess(cronJobModel, emailProcessModel);
+        businessProcessService.startProcess(emailProcessModel);
+
+        cronJobModel.setLastStartTime(cronJobModel.getStartTime());
+        modelService.save(cronJobModel);
+        return new PerformResult(CronJobResult.SUCCESS, CronJobStatus.FINISHED);
+    }
+
+    private void initProcess(QuestionsUpdatesEmailSendingCronJobModel cronJobModel,
+                             QuestionsUpdatesEmailProcessModel emailProcessModel) {
+
+        emailProcessModel.setCustomer((CustomerModel) userService
+                .getUserForUID(cronJobModel.getRecipientEmailAddress()));
+
+        Date lastEndTime = Optional.ofNullable(cronJobModel.getLastStartTime()).orElse(new Date());
```

Comment:
_new Date()_ here as a fallback means that this job will never obtain any questions existing in the database before launching the cronjob for the first time

Reply:
Yes, that was the idea. Or it would be better if I specify some special date? For example, 1 week ago from the current. Or in that case I should obtain all existing questions from db?